### PR TITLE
Example: Dataflow BigQuery to AlloyDB

### PR DESCRIPTION
**Pull request by jogando**
Add an example that shows how to create a Dataflow pipeline that moves data from BigQuery to AlloyDB

### CODE REVIEW

File: examples/dataflow-bigquery-to-alloydb/.gitignore [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# pyenv
+.python-version
+
+# pipenv
+#   According to pypa/pipenv#598, it is recommended to include Pipfile.lock in version control.
+#   However, in case of collaboration, if having platform-specific dependencies or dependencies
+#   having no cross-platform support, pipenv may install dependencies that don't work, or not
+#   install all needed dependencies.
+#Pipfile.lock
```
Comment (shakeebshams) [inline examples/dataflow-bigquery-to-alloydb/.gitignore:83]:
```suggestion
# pipfile.lock
```

File: examples/dataflow-bigquery-to-alloydb/main.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        help='AlloyDB Port')
+    parser.add_argument('--alloydb_database',
+                        dest='alloydb_database',
+                        required=True,
+                        help='AlloyDB Database name')
+    parser.add_argument('--alloydb_table',
+                        dest='alloydb_table',
+                        required=True,
+                        help='AlloyDB table name')
+    parser.add_argument(
```
Comment (shakeebshams) [inline examples/dataflow-bigquery-to-alloydb/main.py:63]: this statement can be completed like the following to maintain consistency

```suggestion
    parser.add_argument('--bq_query',
                                         dest='bq_query',
                                         required=True,
        help='Query to be executed by BigQuery for extracting the source data')
```

File: examples/dataflow-bigquery-to-alloydb/main.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    known_args, pipeline_args = parser.parse_known_args(argv)
+
+    pipeline_options = PipelineOptions(pipeline_args)
+    pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+
+    with beam.Pipeline(options=pipeline_options) as p:
+        coders.registry.register_coder(EthereumTokenTransfersRowType,
+                                       coders.RowCoder)
+
+        (p | 'ReadFromBigQuery' >> beam.io.ReadFromBigQuery(
```
Comment (shakeebshams) [inline examples/dataflow-bigquery-to-alloydb/main.py:77]: Just a suggestion, but this snippet of code isnt very readable
I would rewrite it as such:
```
with beam.Pipeline(options=pipeline_options) as pipeline:
        ( 
            pipeline
            | 'Read Data From BigQuery' >> beam.io.ReadFromBigQuery(
                query=known_args.bq_query, 
                use_standard_sql=True
            )
            | 'Convert BigQuery Rows to Beam Rows' >> beam.Map(lambda bq_row: beam.Row(
                from_address=bq_row['from_address'],
                to_address=bq_row['to_address'],
                value=bq_row['value'],
                block_timestamp=bq_row['block_timestamp'].isoformat()
            )).with_output_types(EthereumTokenTransfersRowType)
            | 'Write Data to JDBC' >> WriteToJdbc(
                driver_class_name='org.postgresql.Driver',
                table_name=known_args.alloydb_table,
                jdbc_url=jdbc_url,
                username=known_args.alloydb_username,
                password=known_args.alloydb_password,
                connection_properties='stringtype=unspecified'
            )
        )

```

**Review by shakeebshams — COMMENTED** [review]
a few suggestions

File: examples/dataflow-bigquery-to-alloydb/main.py (hunk 2 above)
Comment (jogando) [reply examples/dataflow-bigquery-to-alloydb/main.py:63]: Yes, I agree with your statement, but if I implement this change, then the linter implemented in this repository fails.
`python3 -m yapf --diff -r --style google examples/dataflow-bigquery-to-alloydb/main.py`

File: examples/dataflow-bigquery-to-alloydb/main.py (hunk 3 above)
Comment (jogando) [reply examples/dataflow-bigquery-to-alloydb/main.py:77]: Yes, I agree with your statement, but if I implement this change, then the linter implemented in this repository fails.
`python3 -m yapf --diff -r --style google examples/dataflow-bigquery-to-alloydb/main.py`

**Review by shakeebshams — APPROVED** [review]
LGTM

File: examples/dataflow-bigquery-to-alloydb/main.py (hunk 2 above)
Comment (shakeebshams) [reply examples/dataflow-bigquery-to-alloydb/main.py:63]: ack

File: examples/dataflow-bigquery-to-alloydb/main.py (hunk 3 above)
Comment (shakeebshams) [reply examples/dataflow-bigquery-to-alloydb/main.py:77]: ack

**Review by agold-rh — APPROVED** [review]

**Pull request merged**